core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -64,99 +65,71 @@
 public final class Configuration {
   private static final Logger LOG = LoggerFactory.getLogger(Configuration.class);
 
-  /** The source of a configuration property. */
-  public enum Source {
-    DEFAULT,
-    HADOOP_CONF,
-    SITE_PROPERTY,
-    SYSTEM_PROPERTY,
-    UNKNOWN,
-  }
-
   /** Regex string to find "${key}" for variable substitution. */
   private static final String REGEX_STRING = "(\\$\\{([^{}]*)\\})";
   /** Regex to find ${key} for variable substitution. */
   private static final Pattern CONF_REGEX = Pattern.compile(REGEX_STRING);
-  /** Map of properties. */
-  private static final ConcurrentHashMap<String, String> PROPERTIES = new ConcurrentHashMap<>();
-  /** Map of property sources. */
-  private static final ConcurrentHashMap<PropertyKey, Source> SOURCES = new ConcurrentHashMap<>();
+  /** Source of the truth of all property values (default or customized). */
+  private static AlluxioProperties sProperties;
+  /** Path of site properties file. */
   private static String sSitePropertyFile;
+  /** Map from source to properties. */
+  private static Map<Source, Map<?, ?>> sSources = Maps.newHashMap();
```

Comment:
It's a little weird that this hangs around here even though it isn't a source of truth, perhaps we could push the source management logic into `AlluxioProperties`? That would fit with the separation we discussed where `AlluxioProperties` holds the config values while `Configuration` builds extra functionality (validation, type conversion) on top of `AlluxioProperties`

Reply:
Removed from this class to keep this Configuration class involves minimally in different sources.